packages/provider/context/index.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component, createContext } from 'react';
+import Fallback from '../fallback';
+
+const { Provider, Consumer } = createContext({
+  modifiers: () => ([]),
+  getItem: [],
+  Fallback
+});
+
+/**
+ * Flatten the consumer and receiver data into a single array.
+ *
+ * @param {Array|Function} consumer
+ * @param {Array|Function} receiver
+ * @returns {Array}
+ * @public
+ */
+function flatten(consumer, receiver) {
+  return [].concat(consumer).concat(receiver).filter((value, index, all) => {
+    return value && all.indexOf(value) === index;
+  });
+}
+
+/**
+ * Prosumer is a provider combined with a consumer, so you can inherit from
+ * for your provider.
+ *
+ * @param {Object} props The props.
+ * @public
+ */
+function ProSumer({ children, fallback, getItem, modifiers }) {
+  return (
+    <Consumer>
+    {
+      (context) => (
+        <Provider value={{
+          Fallback: fallback || context.Fallback || Fallback,
+          modifiers: modifiers || context.modifiers,
+          getItem: flatten(getItem, context.getItem)
+        }}>
+          { children }
+        </Provider>
+      )
+    }
+    </Consumer>
+  );
+}
+
+/**
+ * Wraps the a given component with the Consumer and applies the Context as
+ * props.
+ *
+ * @param {Component} Component The component that needs to receive the context
+ * @returns {Component} The Consumer wrapped component.
+ * @public
+ */
+function withContext(Component) {
+  const forward = React.forwardRef((props, ref) => (
```

Comment:
Consider using [`addhoc`](https://github.com/godaddy/addhoc) to take care of this sort of thing for you.

Reply:
I don't see how adding [4.3kb of additional bytes](https://bundlephobia.com/result?p=addhoc@1.0.2) is going to be an improvement here. In fact, using `addhoc` will make it harder to debug because it adds an additional component layer in the render tree that wouldn't be needed, and doesn't set the correct displayName for forwarded refs.

Reply:
Fair enough. I understand that the statics hoisting is not needed here.

Regarding the display name issue, can you please either file an issue or a PR on addhoc to fix?